getting_started/scripting/gdscript/gdscript_basics.rst

```diff
@@ -1463,104 +1463,137 @@ freed.
 Signals
 ~~~~~~~
 
-Signals are a way to send notification messages from an object that
-other objects can listen to in a generic way. Create custom signals for
-a class using the ``signal`` keyword.
+Signals are a tool to emit messages from an object that other objects can react
+to. To create custom signals for a class, use the ``signal`` keyword.
 
 ::
 
-   # Signal with no arguments
-   signal your_signal_name
+   extends Node
 
-   # Signal with two arguments
-   signal your_signal_name_with_args(a, b)
+   # A signal named health_depleted
+   signal health_depleted
 
-These signals may be connected to methods in the same manner as you connect
-built-in signals of nodes such as :ref:`class_Button` or :ref:`class_RigidBody`.
+   # A signal that forwards two arguments
+   signal health_changed(old_value, new_value)
 
-Here's an example that creates a custom signal in one script and connects
-the custom signal to a method in a separate script, using the
-:ref:`Object.connect() <class_Object_method_connect>` method:
+   func took_damage(amount):
+   emit_signal()
 
-::
+.. note:: 
+
+   Signals are a `Callback
+   <https://en.wikipedia.org/wiki/Callback_(computer_programming)>`
+   mechanism. They also fill the role of Observers, a common programming
+   pattern. For more information, read the `Observer
+   tutorial<http://gameprogrammingpatterns.com/observer.html>`_ in the
+   Game Programming Patterns ebook.
 
-   # your_notifier.gd
+You can connect these signals to methods the same way you connect built-in
+signals of nodes like :ref:`class_Button` or :ref:`class_RigidBody`.
 
-   signal data_found
+For an example of when signals can be useful, let's consider we have a game
+where a character can take damage from enemies. We want a life bar on screen to
+react to health changes with an animation, but we want to keep the user
+interface separate from the player in our scene tree.
 
-   var your_data = 42
+In our ``Character.gd`` script, we define a ``health_changed`` signal and emit
+it with :ref:`Object.emit_signal() <class_Object_method_emit_signal>`, and from
+a ``Game`` node higher up our scene tree, we connect it to the ``Lifebar`` using
+the :ref:`Object.connect() <class_Object_method_connect>` method:
 
 ::
 
-   # your_handler.gd
+    # Character.gd
 
-   func your_handler():
-      print("Your handler was called!")
+    ...
+    signal health_changed(new_value)
 
-::
+    func take_damage(amount):
+        var old_health = health
+        health -= amount
 
-   # your_game.gd
+        # We emit the health_changed signal every time the
+        # character took damage
+        emit_signal('health_changed', old_health, health)
```

Comment:
I feel double quotes are used more often than single quotes in GDScript. That being said, I cannot find anything about it in the styleguide (which uses both single and double quotes).